websites/G/galaxy/presence.ts

```diff
@@ -0,0 +1,432 @@
+const presence = new Presence({
+		clientId: "1122404044829380739",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/KQvYGoh.png",
+			startTimestamp: browsingTimestamp,
+		},
+		{ href, pathname } = document.location,
+		path = pathname.slice(1).split("/"),
+		[showTimestamp, showButtons] = await Promise.all([
+			presence.getSetting<boolean>("timestamp"),
+			presence.getSetting<boolean>("buttons"),
+		]);
+
+	switch (path[0]) {
+		case "": {
+			presenceData.details = "Viewing the front page";
+			break;
+		}
+		// Game pages
+		case "play": {
+			if (path[1]) {
+				const gameName =
+						document.querySelector<HTMLHeadingElement>(
+							"#game-title"
+						)?.textContent,
+					currentLevel = document.querySelector<HTMLSpanElement>(
+						"#progress > .level > span"
+					)?.textContent,
+					// "N / D"
+					xpRatio = document.querySelector<HTMLSpanElement>(
+						"#progress > .xp > span"
+					)?.textContent;
+
+				presenceData.details = gameName
+					? `Playing ${gameName}`
+					: "Playing a game";
+				if (currentLevel && xpRatio) {
+					const [currentXP, neededXP] = xpRatio.split("/").map(n => n.trim());
+
+					presenceData.state = `Level ${currentLevel} | ${currentXP}/${neededXP} XP`;
+				}
+				presenceData.buttons = [
+					{
+						label: "Play game",
+						url: href,
+					},
+				];
+			}
+			break;
+		}
+		case "edit": {
+			if (path[1]) {
+				const gameName = document.querySelector<HTMLAnchorElement>(
+					"#container > h1 > a"
+				)?.textContent;
+
+				presenceData.details = gameName
+					? `Editing ${gameName}`
+					: "Editing a game";
+				presenceData.buttons = [
+					{
+						label: "Play game",
+						url: href,
+					},
```

Comment:
Words in button labels should be capitalized (`Play Game`, `Read Comments`, etc.)